consensus/polybft/validator_set.go

```diff
@@ -323,6 +340,27 @@ func (v *validatorSet) rescalePriorities(diffMax int64) error {
 	return nil
 }
 
+// HasQuorum determines if there is quorum of enough signers reached,
+// based on its voting power and quorum size
+func (v validatorSet) HasQuorum(signers []types.Address) bool {
+	return v.calculateVotingPower(signers) >= v.quorumSize
+}
+
+// checks if submitted signers have reached prepare quorum
+func (v validatorSet) HasPrepareQuorum(signers []types.Address) bool {
+	return v.calculateVotingPower(signers) >= v.quorumSize-1
```

Comment:
I'd say we should substract local node's voting power here instead of 1. Because it would be wrong to assume that local node has `votingPower=1`.

```go
return v.calculateVotingPower(signers) >= v.quorumSize-<localNodeVotingPower>
```